test/suite/wttest.py

```diff
@@ -42,9 +42,31 @@
     import unittest
 
 from contextlib import contextmanager
-import errno, glob, os, re, shutil, sys, time, traceback
+import errno, glob, os, re, shutil, sys, threading, time, traceback
 import wiredtiger, wtscenario, wthooks
 
+# Use as "with timeout(seconds): ....". Argument of 0 means no timeout,
+# and only available (with non-zero argument) on Unix systems.
+class timeout(object):
+    def __init__(self, seconds=0):
+        self.seconds = seconds
+        self.prev_handler = None
+
+    def signal_handler(self, signum, frame):
+        raise(TimeoutError('time for test exceeded {} seconds'.format(self.seconds)))
+
+    def __enter__(self):
+        if self.seconds != 0:
+            import signal     # This will fail on non-Unix systems.
```

Comment:
Is there a way to fail more gracefully on non-Posix systems?  I'm imagining that on Windows, for example, we would simply get a generic python error about failing to find/import signal, and a naive user might waste a bunch of time trying to figure out where/how to add that library to their system. Could we instead generate an error message here (or in parsing the timeout argument) that explicitly tells the user this argument isn't supported on non-Posix systems?

Reply:
fixed.